gen/enum_test.go

```diff
@@ -335,6 +335,28 @@ func TestUnmarshalTextReturnsErrorOnInvalidValue(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestMarshalTextReturnsValue(t *testing.T) {
+	tests := []struct {
+		title string
+		e     te.EnumDefault
+		text  string
+	}{
+		{`0 <-> Foo`, te.EnumDefaultFoo, "Foo"},
+		{`1 <-> Bar`, te.EnumDefaultBar, "Bar"},
+		{`2 <-> Baz`, te.EnumDefaultBaz, "Baz"},
+		{`42 <-> 42`, 42, "42"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.title, func(t *testing.T) {
+			b, err := tt.e.MarshalText()
+			if assert.NoError(t, err, "unable to marshal enum") {
+				assert.Equal(t, []byte(tt.text), b, "text output doesn't match")
+			}
```

Comment:
I think it would be valuable to verify that values are round-tripped correctly
here: call UnmarshalText on the result to verify that you get back what the
input was. (In which case this whole test can be renamed to
`TestTextMarshaler`.)

Reply:
Sounds good, Unmarshaling doesn't seem to support unknown values so I removed the 42 test.

Reply:
I would add a separate test case for just unknown value marshaling while keeping this for known values.